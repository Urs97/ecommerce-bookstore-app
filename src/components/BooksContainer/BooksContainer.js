import React from 'react';
import './BooksContainer.scss';
import Book from '../Book/Book';

const BooksContainer = ({ bookData }) => {

    const books = bookData.map(book => <Book key={book.key} book={book}/>);

    let content = <section className="no-books">No books found.</section>

    if(books.length !== 0) {
        content = (
            <section className="books-container-section">
                <section className="books">
                    {books}
                </section>
            </section>
        );
    };

    return content;
};

export default BooksContainer;