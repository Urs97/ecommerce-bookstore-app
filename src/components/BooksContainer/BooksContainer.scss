@import '../../styles/shared.scss';

.books-container-section {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    background-color: $background-grey;
    padding: $vertical-content-padding $horizontal-content-padding;
    color: $text-color;

    .books {
        width: 100%;
        display: grid;
        grid-template-columns: repeat(auto-fit, minmax(12em, 1fr));
        gap: 4em;
    }
}

.no-books {
    font-size: 1.5em;
    text-align: center;
}

@media (max-width: 1100px) {
    .books-container-section {
        padding: $vertical-content-padding $horizontal-content-padding-1100px;
    }
}

@media (max-width: 950px) {
    .books-container-section {
        padding: $vertical-content-padding $horizontal-content-padding-950px;
    }
}

@media (max-width: 865px) {
    .books-container-section {
        padding: $vertical-content-padding $horizontal-content-padding-865px;
    }
}