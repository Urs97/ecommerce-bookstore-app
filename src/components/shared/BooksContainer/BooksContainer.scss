@import '../../../styles/shared.scss';

.books-container-section {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #F4F7F9;
    padding: $vertical-content-padding $horizontal-content-padding;
    color: $text-color;

    & h1 {
        font-family: 'Open Sans', sans-serif;
        font-weight: 600;
        margin: 0;
        margin-bottom: $vertical-content-padding;
        font-size: 2.8em;
    }

    & a {
        margin-top: $vertical-content-padding;
    }

    .books {
        width: 100%;
        display: grid;
        grid-template-columns: repeat(auto-fit, minmax(14em, 1fr));
        gap: 4em;
    }
}