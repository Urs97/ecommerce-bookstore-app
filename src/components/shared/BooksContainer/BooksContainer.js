import React from 'react'
import './BooksContainer.scss';
import { Link } from 'react-router-dom';
import Book from '../Book/Book';

function BooksContainer() {
    return (
        <section className="books-container-section">
            <h1>Books Bestsellers</h1>
            <section className="books">
                <Book />
                <Book />
                <Book />
                <Book />
                <Book />
                <Book />
                <Book />
                <Book />
            </section>
            <Link to='/store' className="button">View More</Link>
        </section>
    )
}

export default BooksContainer;