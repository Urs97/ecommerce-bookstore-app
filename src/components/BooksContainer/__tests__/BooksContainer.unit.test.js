import React from 'react';
import {shallow} from 'enzyme';
import BooksContainer from '../BooksContainer';

describe('BooksContainer', () => {
    test('renders empty BooksContainer', () => {
        const wrapper = shallow(
            <BooksContainer />
        );
        expect(wrapper).toMatchSnapshot();
    });
});