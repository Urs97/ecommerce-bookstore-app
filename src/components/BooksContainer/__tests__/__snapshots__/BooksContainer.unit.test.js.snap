// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksContainer renders BooksContainer 1`] = `
<section
  className="books-container-section"
>
  <section
    className="books"
  >
    <Book
      book={
        Object {
          "authors": Array [
            Object {
              "key": "/authors/OL25342A",
              "name": "Marry Shelley",
            },
          ],
          "cover_edition_key": "OL6147070M",
          "key": "/works/OL450125W",
          "price": Array [
            94,
            99,
          ],
          "title": "Frankenstein",
        }
      }
      key="/works/OL450125W"
    />
    <Book
      book={
        Object {
          "authors": Array [
            Object {
              "key": "/authors/OL13066A",
              "name": "H. G. Wells",
            },
          ],
          "cover_edition_key": "OL8000344M",
          "key": "/works/OL52267W",
          "price": Array [
            75,
            65,
          ],
          "title": "The Time Machine",
        }
      }
      key="/works/OL52267W"
    />
  </section>
</section>
`;

exports[`BooksContainer renders empty BooksContainer 1`] = `
<section
  className="no-books"
>
  No books found.
</section>
`;
